verifyconcor/src/main/java/uk/gov/justice/laadces/verifyconcor/Runner.java

```diff
@@ -0,0 +1,131 @@
+import uk.gov.justice.laadces.verifyconcor.repository.AnomalyRepository;
+import uk.gov.justice.laadces.verifyconcor.repository.ConcorContribution;
+import uk.gov.justice.laadces.verifyconcor.repository.ConcorContributionRepository;
+import uk.gov.justice.laadces.verifyconcor.repository.StatisticRepository;
+import uk.gov.justice.laadces.verifyconcor.service.JSONTransformService;
+import uk.gov.justice.laadces.verifyconcor.service.JSONValidationService;
+import uk.gov.justice.laadces.verifyconcor.service.XMLComparisonService;
+import uk.gov.justice.laadces.verifyconcor.service.XMLTransformService;
+
+import java.util.List;
+
+/**
+ * ApplicationRunner class that examines the data in the database.
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+class Runner implements ApplicationRunner {
+    private final AnomalyRepository anomalyRepository;
+    private final ConcorContributionRepository concorRepository;
+    private final StatisticRepository statisticRepository;
+
+    private final JSONTransformService jsonTransformService;
+    private final JSONValidationService jsonValidationService;
+    private final XMLComparisonService xmlComparisonService;
+    private final XMLTransformService xmlTransformService;
+
+    private static final long LOW_ID = 300_000_000L;
+    private static final long HIGH_ID = 499_999_999L;
+    private static final int BATCH_SIZE = 1000;
+
+    /**
+     * Main runner method, just iterates over the database table, calling #examineConcorContribution() for each row.
+     * Handles limiting the data size and paging efficiently, with some progress-logging to the application log.
+     *
+     * @param args ApplicationArguments ignored.
+     * @throws Exception if an error occurs.
+     */
+    @Override
+    public void run(ApplicationArguments args) throws Exception {
+        final int recs = concorRepository.countByIdBetween(LOW_ID, HIGH_ID);
+        log.info("There are {} rows with {} <= id <= {}", recs, LOW_ID, HIGH_ID);
+        long nextLowId = LOW_ID;
+        int recsLoaded = 0;
+        log.info("-----BEGIN-----");
+        final long t0 = System.currentTimeMillis();
+
+        Slice<ConcorContribution> slice = concorRepository.findByIdBetweenOrderById(nextLowId, HIGH_ID, Pageable.ofSize(BATCH_SIZE));
+        List<ConcorContribution> content = slice.getContent();
+        int contentLen = content.size();
+        recsLoaded += contentLen;
+        long sliceLowId = contentLen > 0 ? content.get(0).id() : nextLowId;
+        long sliceHighId = contentLen > 0 ? content.get(contentLen - 1).id() : HIGH_ID;
+        nextLowId = sliceHighId + 1;
+        log.info("Loaded {} rows with {} <= id <= {}, {} / {} rows loaded, {} ms", contentLen, sliceLowId, sliceHighId, recsLoaded, recs, System.currentTimeMillis() - t0);
+        content.forEach(this::examineConcorContribution);
+
+        while (slice.hasNext() && contentLen > 0 && nextLowId <= HIGH_ID) {
```

Comment:
Originally the logic was slightly different the first time round the loop because I used repository pagination. Now that I'm using Slices and Ids I can do this - well spotted. Will refactor.